pulp_container/app/registry_api.py

```diff
@@ -1426,7 +1425,7 @@ def put(self, request, path, pk):
         except models.Manifest.DoesNotExist:
             raise ManifestNotFound(reference=pk)
 
-        signature_payload = request.META["wsgi.input"].read(SIGNATURE_PAYLOAD_MAX_SIZE)
+        signature_payload = request.META["wsgi.input"].read()
```

Comment:
It surprises me that this uses such a low level implementation. If I read the code right, `request` is a `rest_framework.request.Request` which has a method `request.stream` to get it as `io.BytesIO`. Isn't that a much more portable interface? Then I think you can actually use `signature_dict = json.load(request.stream)` to save copying the full response into memory an additional time.

And if you want to guard it, you can inspect the length explicitly prior to loading and send HTTP 413. See `_load_stream()` for how they read the length:
https://github.com/encode/django-rest-framework/blob/f593f5752c45e06147231bbfd74c02384791e074/rest_framework/request.py#L297-L314

My primary reason to avoid `wsgi.input` (and this isn't the only occurrence) is that it blocks you from moving to ASGI and eventually more performant application servers.

Reply:
@git-hyagi, can you also address this comment, please? It is worth the effort.